src/ucp/api/ucp_def.h

```diff
@@ -234,5 +234,15 @@ typedef void (*ucp_send_callback_t)(void *request, ucs_status_t status);
 typedef void (*ucp_tag_recv_callback_t)(void *request, ucs_status_t status,
                                         ucp_tag_recv_info_t *info);
 
+/**
+ * @ingroup UCP_WORKER
+ * @brief UCP worker wakeup events mask.
+ *
+ * The enumeration allows specifying which events are expected on wakeup
+ */
+typedef enum ucp_wakeup_event_types {
+    UCP_WAKEUP_TX_COMPLETION   = UCS_BIT(0), /**< TX-completions */
+    UCP_WAKEUP_RX_AM           = UCS_BIT(1)  /**< RX-completions */
```

Comment:
Yes, it's visible and optional. It's signaled for every corresponding TX/RX completions the same way as UCT does. Really, there is nothing new, just filtering from UCP level. This is not related directly with UCP requests. See conversation in https://github.com/openucx/ucx/issues/1062